cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -1060,32 +1057,124 @@ func (s *Server) AddPackageRepository(ctx context.Context, request *corev1.AddPa
+	}
+	log.Infof("+helm GetPackageRepositoryDetail '%s' in context [%v]", repoRef.Identifier, repoRef.Context)
+
+	cluster := repoRef.GetContext().GetCluster()
+	namespace := repoRef.GetContext().GetNamespace()
+	name := repoRef.GetIdentifier()
+
+	log.Infof("+helm GetPackageRepositoryDetail (cluster=%s, namespace=%s, name=%s)", cluster, namespace, name)
+
+	// Retrieve repository
+	appRepo, caCertSecret, authSecret, err := s.getAppRepoAndRelatedSecrets(ctx, cluster, name, namespace)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Unable to retrieve AppRepository '%s/%s' due to [%v]", namespace, name, err)
+	}
+
+	// Map to target struct
+	repositoryDetail, err := s.mapToPackageRepositoryDetail(appRepo, cluster, namespace, caCertSecret, authSecret)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unable to convert the AppRepository: %v", err)
+	}
+
+	// response
+	response := &corev1.GetPackageRepositoryDetailResponse{
+		Detail: repositoryDetail,
+	}
+
+	return response, nil
 }
 
 func (s *Server) GetPackageRepositorySummaries(ctx context.Context, request *corev1.GetPackageRepositorySummariesRequest) (*corev1.GetPackageRepositorySummariesResponse, error) {
-	// TODO(rcastelblanq) WIP
-	return nil, nil
+	log.Infof("+helm GetPackageRepositorySummaries [%v]", request)
+	if request.GetContext() == nil || request.GetContext().GetCluster() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no valid context provided")
+	}
+
+	if summaries, err := s.repoSummaries(ctx, request.GetContext().GetCluster(), request.GetContext().GetNamespace()); err != nil {
+		return nil, err
+	} else {
+		return &corev1.GetPackageRepositorySummariesResponse{
+			PackageRepositorySummaries: summaries,
+		}, nil
+	}
 }
 
 func (s *Server) UpdatePackageRepository(ctx context.Context, request *corev1.UpdatePackageRepositoryRequest) (*corev1.UpdatePackageRepositoryResponse, error) {
-	// TODO(rcastelblanq) WIP
-	return nil, nil
+	if request == nil || request.PackageRepoRef == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no request PackageRepoRef provided")
+	}
+	repoRef := request.GetPackageRepoRef()
+	if repoRef.GetContext() == nil || repoRef.GetContext().GetCluster() == "" || repoRef.GetContext().GetNamespace() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no valid context provided")
+	}
+	log.Infof("+helm UpdatePackageRepository '%s' in context [%v]", repoRef.Identifier, repoRef.Context)
+
+	// Get Helm-specific values
+	var customDetails *helmv1.RepositoryCustomDetails
+	if request.CustomDetail != nil {
+		customDetails = &helmv1.RepositoryCustomDetails{}
+		if err := request.CustomDetail.UnmarshalTo(customDetails); err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "customDetails could not be parsed: [%v]", request.CustomDetail)
+		}
+		log.V(4).Infof("+helm upgrade repo %s customDetails [%v]", repoRef.Identifier, customDetails)
+	}
+
+	helmRepo := &HelmRepository{
```

Comment:
Using this custom `HelmRepository` to avoid having 8 parameters in functions signatures :)